contributors/design-proposals/multicluster/cluster-registry/project-design-and-plan.md

```diff
@@ -0,0 +1,335 @@
+-   [Test strategy](#test-strategy)
+-   [Milestones and timelines](#milestones-and-timelines)
+    -   [Alpha (targeting late Q4 '17)](#alpha-targeting-late-q4-'17)
+    -   [Beta (targeting mid Q1 '18)](#beta-targeting-mid-q1-'18)
+    -   [Stable (targeting mid Q2 '18)](#stable-targeting-mid-q2-'18)
+    -   [Later](#later)
+
+## Background
+
+SIG-multicluster has identified a cluster registry as being a key enabling
+component for multi-cluster use cases. The goal of the SIG in this project is
+that the API defined for the cluster registry become a standard for
+multi-cluster tools. The API design is being discussed in
+[a separate doc](https://drive.google.com/a/google.com/open?id=1Oi9EO3Jwtp69obakl-9YpLkP764GZzsz95XJlX1a960).
+A working prototype of the cluster registry has been assembled in
+[a new repository](https://github.com/kubernetes/cluster-registry).
+
+## Goal
+
+This document intends to lay out an initial plan for moving the cluster registry
+from the prototype state through alpha, beta and eventually to a stable release.
+
+## Technical requirements
+
+These requirements are derived mainly from the output of the
+[August 9th multi-cluster SIG meeting](https://docs.google.com/document/d/11cB3HK67BZUb7aNOCpQK8JsA8Na8F-F_6bFu77KrKT4/edit#heading=h.cuvqls7pl9qc).
+However, they also derive (at least indirectly) from the results of the
+[SIG F2F meeting ](https://docs.google.com/document/d/1HkVBSm9L9UJC2f3wfs_8zt1PJmv6iepdtJ2fmkCOHys/edit)
+that took place earlier, as well as the use cases presented by various parties
+at that meeting.
+
+### Alpha
+
+-   [API] Provides an HTTP server with a Kubernetes-style API for CRUD
+    operations on a registry of clusters
+    -   Provides a way to filter across clusters by label
+    -   Supports annotations
+-   [API] Provides information about each cluster's API server's
+    location/endpoint
+-   [API] Provides the ability to assign user-friendly names for clusters
+-   [API] Provides pointers to authentication information for clusters' API
+    servers
+    -   This implies that it does not support storing credentials directly
+-   [Implementation] Supports both independent and aggregated deployment models
+    -   Supports delegated authn/authz in aggregated mode
+    -   Supports integration into Federation via aggregated deployment model
+
+### Beta
+
+-   Supports providing a flat file of clusters for storage and later use
+    -   This may be provided by the ecosystem rather than the registry API
+        implementation directly
+-   Supports independent authz for reading/writing clusters
+-   `kubectl` integration with the cluster registry API server is first-class
+    and on par with kubectl integration with core Kubernetes APIs
+-   Supports grouping of clusters
+-   Supports specifying and enforcing read and/or write authorization for groups
+    of clusters in the registry
+-   Working Federation integration
+-   Supports status from active controllers
+
+### Later {#later}
```

Comment:
{#later} ?

Reply:
Whoops! That's an artifact from the markdown conversion. Thanks for catching that!